subsys/logging/CMakeLists.txt

```
@@ -14,6 +14,15 @@ if(NOT CONFIG_LOG_MINIMAL)
     log_backend_uart.c
   )
 
+  zephyr_sources_ifdef(
+    CONFIG_LOG_BACKEND_FS
+    log_backend_fs.c
+  )
+
+  zephyr_include_directories(
+    ${ZEPHYR_BASE}/../modules/fs/littlefs
+  )
```

```suggestion
  if(CONFIG_LOG_BACKEND_FS AND CONFIG_FILE_SYSTEM_LITTLEFS)
    zephyr_link_libraries(LITTLEFS)
  endif()
```

The reason for the `CONFIG_FILE_SYSTEM_LITTLEFS` test was because I could only see `LOG_BACKEND` is dependent on `FILE_SYSTEM` and not a specific `FILE_SYSTEM` implementation.
https://github.com/zephyrproject-rtos/zephyr/blob/9f7c0f3c0ef399ca5afae486aca19e694bda6eac/subsys/logging/Kconfig#L285

---

Please let me know if this doesn't work.

---

if making `LOG_BACKEND_FS` depends on `FILE_SYSTEM_LITTLEFS`, then this can be simplified further into:

```suggestion
  zephyr_link_libraries_ifdef(CONFIG_LOG_BACKEND_FS LITTLEFS)
```

---

> Please let me know if this doesn't work.

It works fine. I used the last version
`zephyr_link_libraries_ifdef(CONFIG_LOG_BACKEND_FS LITTLEFS)`
as the LittleFS is the only currently tested FS.